src/subdomains/core/payment-link/dto/payment-link-dto.mapper.ts

```diff
@@ -39,4 +40,40 @@ export class PaymentLinkDtoMapper {
       }
     );
   }
+
+  static toRecipientDto(paymentLink: PaymentLink): PaymentLinkRecipientDto | undefined {
+    if (paymentLink.hasRecipient) {
+      return {
+        name: paymentLink.name,
+        address: {
+          street: paymentLink.street,
+          houseNumber: paymentLink.houseNumber,
+          zip: paymentLink.zip,
+          city: paymentLink.city,
+          country: paymentLink.country?.name,
+        },
+        phone: paymentLink.phone,
+        mail: paymentLink.mail,
+        website: paymentLink.website,
+      };
+    }
+
+    const userData = paymentLink.route?.userData;
+
+    if (userData) {
+      return {
+        name: `${userData.firstname} ${userData.surname}`,
+        address: {
```

Comment:
`name` only if there is a firstname and surname (will be `undefined undefined` otherwise 😉)